spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/PubsubAdmin.java

```diff
@@ -0,0 +1,196 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub;
+
+import java.util.List;
+
+import com.google.cloud.pubsub.v1.PagedResponseWrappers;
+import com.google.cloud.pubsub.v1.SubscriptionAdminClient;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.common.collect.Lists;
+import com.google.pubsub.v1.ProjectName;
+import com.google.pubsub.v1.PushConfig;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.util.Assert;
+
+/**
+ * Pub/Sub admin utility that creates new topics and subscriptions on Google Cloud Pub/Sub.
+ *
+ * @author João André Martins
+ */
+public class PubsubAdmin {
+
+	private final String projectId;
+
+	private TopicAdminClient topicAdminClient;
+
+	private SubscriptionAdminClient subscriptionAdminClient;
+
+	/** Default inspired in the subscription creation web UI. */
+	private static final int DEFAULT_ACK_DEADLINE = 10;
+
+	public PubsubAdmin(GcpProjectIdProvider projectIdProvider, TopicAdminClient topicAdminClient,
+			SubscriptionAdminClient subscriptionAdminClient) {
+		Assert.notNull(projectIdProvider, "The project ID provider can't be null.");
+		Assert.notNull(topicAdminClient, "The topic administration client can't be null");
+		Assert.notNull(subscriptionAdminClient,
+				"The subscription administration client can't be null");
+
+		this.projectId = projectIdProvider.getProjectId();
+		Assert.hasText(this.projectId, "The project ID can't be null or empty.");
+		this.topicAdminClient = topicAdminClient;
+		this.subscriptionAdminClient = subscriptionAdminClient;
+	}
+
+	/**
+	 * Creates a new topic on Google Cloud Pub/Sub.
+	 *
+	 * @param topicName the name for the new topic
+	 * @return the created topic
+	 */
+	public Topic createTopic(String topicName) {
+		Assert.hasText(topicName, "No topic name was specified.");
+
+		return this.topicAdminClient.createTopic(TopicName.create(this.projectId, topicName));
+	}
+
+	/**
+	 * Deletes a topic from Google Cloud Pub/Sub.
+	 *
+	 * @param topicName the name of the topic to be deleted
+	 */
+	public void deleteTopic(String topicName) {
+		Assert.hasText(topicName, "No topic name was specified.");
+
+		this.topicAdminClient.deleteTopic(TopicName.create(this.projectId, topicName));
+	}
+
+	/**
+	 * Returns every topic in a project.
+	 *
+	 * <p>If there are multiple pages, they will all be merged into the same result.
+	 */
+	public List<Topic> listTopics() {
+		PagedResponseWrappers.ListTopicsPagedResponse topicListPage =
+				this.topicAdminClient.listTopics(ProjectName.create(this.projectId));
+
+		return Lists.newArrayList(topicListPage.iterateAll());
```

Comment:
It would make sense for a local variable containing the result of `iterateAll()` to be called `topicsList`, not the `topicAdminClient.listTopics()` result. Also, the object type is `ListTopicsPagedResponse`, which also hints to it being a page, rather than the whole thing.